### PR TITLE
BugFix - 39649 - Not Able To Learn Shadow DOM Elements

### PR DESCRIPTION
**Pull request by IamRanjeetSingh**
Thank you for your contribution.
Before submitting this PR, please make sure:

- [ ] PR description and commit message should describe the changes done in this PR
- [ ] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [ ] Latest Code from master or specific release branch is merged to your branch
- [ ] No unwanted\commented\junk code is included
- [ ] No new warning upon build solution
- [ ] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve all Codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Update the Help Library document to match any feature changes

## Summary by CodeRabbit

- **New Features**
  - Enhanced HTML element learning process by adding a new parameter for selective node learning.
  - Improved Playwright driver with shadow DOM handling and exception management.
  - Added timeout parameter for capturing screenshots in Playwright Browser Element.

- **Refactor**
  - Namespace updates and minor formatting changes for better code organization.

- **Bug Fixes**
  - Fixed issue with shadow DOM element recognition in Playwright.

- **Chores**
  - Removed unnecessary import statements to clean up the codebase.

### CODE REVIEW

File: Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/POM/POMLearner.cs [hunk 1]
```diff
@@ -161,13 +150,8 @@ private static bool CheckStaticHtmlNodeExclusion(HtmlNode htmlNode)
             return true;
         }
 
-        private static bool FormChildrenElementFilter(HtmlNode htmlNode)
+        private static bool ShouldLearnFormChildNode(HtmlNode htmlNode)
         {
```
Comment (Maheshkale447) [inline Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/POM/POMLearner.cs:154]: add error handling

File: Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Playwright/PlaywrightDriver.cs [hunk 2]
```diff
@@ -24,6 +24,7 @@
 using GingerCore.Actions.VisualTesting;
 using GingerCoreNET.SolutionRepositoryLib.RepositoryObjectsLib.PlatformsLib;
 using amdocs.ginger.GingerCoreNET;
+using Amdocs.Ginger.CoreNET.Drivers.CoreDrivers.Web.Selenium;
```
Comment (Maheshkale447) [inline Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Playwright/PlaywrightDriver.cs:27]: Remove selenium references

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

File: Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Playwright/PlaywrightDriver.cs (hunk 2 above)
Comment (IamRanjeetSingh) [reply Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Playwright/PlaywrightDriver.cs]: handled

File: Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/POM/POMLearner.cs (hunk 1 above)
Comment (IamRanjeetSingh) [reply Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/POM/POMLearner.cs:172]: handled

File: Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/POM/POMLearner.cs [hunk 3]
```diff
@@ -83,64 +78,58 @@ private async Task LearnDocumentElementsAsync(HtmlDocument htmlDocument, IList<E
...
-                        await LearnHtmlNodeChildElements(childNode, shouldLearnElement, learnedElements);
+                        childElement = await CreateHTMLElementInfoAsync(childNode, browserElement);
                     }
-
-                    continue;
                 }
 
-                IBrowserElement? browserElement = await _browserElementProvider.GetElementAsync(eLocateBy.ByXPath, childNode.XPath);
-                if (browserElement == null || !await browserElement.IsVisibleAsync())
+                if (childElement != null && shouldLearnNode(childNode))
```
[the remark this answers is not in the document]
Comment (IamRanjeetSingh) [reply Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/POM/POMLearner.cs]: handled

**Review by Maheshkale447 — APPROVED** [review]

**Pull request merged**